x-pack/platform/plugins/shared/triggers_actions_ui/public/common/expression_items/when.test.tsx

```diff
@@ -74,43 +76,51 @@ describe('when expression', () => {
       />
     );
 
-    expect(wrapper.find('[data-test-subj="whenExpressionSelect"]')).toMatchInlineSnapshot(`
-    <EuiSelect
-      data-test-subj="whenExpressionSelect"
-      fullWidth={true}
-      id="aggTypeField"
-      onChange={[Function]}
-      options={
-        Array [
-          Object {
-            "text": "Test1()",
-            "value": "test1",
-          },
-          Object {
-            "text": "Test2()",
-            "value": "test2",
-          },
-        ]
-      }
-      value="count"
-    />
-    `);
+    // Initially, only the button should be visible
+    expect(screen.getByTestId('whenExpression')).toBeInTheDocument();
+    expect(screen.getByText('Test1()')).toBeInTheDocument();
+
+    // Open the popover to see the select
+    await user.click(screen.getByTestId('whenExpression'));
+
+    await waitFor(() => {
+      expect(screen.getByTestId('whenExpressionSelect')).toBeInTheDocument();
+    });
+
+    const select = screen.getByTestId('whenExpressionSelect');
+    expect(select).toHaveValue('test1');
+
+    // Check that custom options are present in the select
+    const options = screen.getAllByRole('option');
+    expect(options).toHaveLength(2);
+    expect(options[0]).toHaveTextContent('Test1()');
+    expect(options[1]).toHaveTextContent('Test2()');
   });
 
-  it('renders when popover title', () => {
+  it('renders when popover title', async () => {
+    const user = userEvent.setup();
     const onChangeSelectedAggType = jest.fn();
-    const wrapper = shallow(
+    renderWithIntl(
       <WhenExpression aggType={'avg'} onChangeSelectedAggType={onChangeSelectedAggType} />
     );
-    wrapper.simulate('click');
-    expect(wrapper.find('[value="avg"]').length > 0).toBeTruthy();
-    expect(
-      wrapper.contains(
-        <FormattedMessage
-          id="xpack.triggersActionsUI.common.expressionItems.threshold.popoverTitle"
-          defaultMessage="when"
-        />
-      )
-    ).toBeTruthy();
+
+    // Initially, only the button should be visible
+    expect(screen.getByTestId('whenExpression')).toBeInTheDocument();
+    expect(screen.getByText('average()')).toBeInTheDocument();
+
+    // Open the popover
+    await user.click(screen.getByTestId('whenExpression'));
+
+    await waitFor(() => {
+      expect(screen.getByTestId('whenExpressionSelect')).toBeInTheDocument();
+    });
+
+    // Check that the select has the correct value
+    const select = screen.getByTestId('whenExpressionSelect');
+    expect(select).toHaveValue('avg');
+
+    // Check that the popover title is rendered (use getAllByText to handle multiple instances)
+    const whenElements = screen.getAllByText('when');
+    expect(whenElements.length).toBeGreaterThan(0);
```

Comment:
It is weird/unnecessary to put these in variables when they are only used once.

We should be really careful to make sure we dont end up with needlessly bloated test files when doing these migrations.

And why the comments also? the code should speak for itself.